### DISCUSSION TITLE
Can "Input" values be updated from secondary inputs?

### DISCUSSION THREAD

**Discussion by gouldingken** [observablehq/framework#883] [Q&A] [answered]
If I want to update all my inputs based on pre-baked "scenarios" - is there a way to update all values from data? This would allow users to test variables in my model, but be able to go back to predefined scenarios at the click of a button.

e.g. if I have an input like this:
```
const population = view(Inputs.range([0, 10000], {step: 10}));
```

is there a way to modify it from a button?
```js
const reset1 = view(Inputs.button("Reset to 1"));
```

```js
reset1;
population = 555
```

Obviously reassignment like this is not allowed and throws:
`SyntaxError: Assignment to external variable 'population'`

But is there a way to support this use case?

**Comment by mbostock** [comment, marked as answer]
Yes, but you need to declare the input as well as its (reactive) value:

```js
const populationInput = Inputs.range([0, 10000], {step: 10});
const population = view(populationInput);
```

Then you can do something like:

```js
populationInput.value = 250;
populationInput.dispatchEvent(new Event("input", {bubbles: true}));
```

**Reply by gouldingken** [reply]
That works like a charm. Thanks!
